op-proposer/proposer/driver.go

```diff
@@ -404,6 +390,10 @@ func (l *L2OutputSubmitter) sendTransaction(ctx context.Context, output *eth.Out
 			return err
 		}
 	} else {
+		err := l.waitForL1Head(ctx, output.Legacy.HeadL1.Number+1)
+		if err != nil {
+			return err
+		}
```

Comment:
I'm not sure why the proposer waits for the L1 head when proposing L2OO outputs. Would it make sense to remove this entirely?

Reply:
It's a weird case - there's a comment n the `waitForL1Head` method itself but it boils down to the fact that L2OO has some reorg protection so it checks the block hash you supply matches, but since geth now uses the _current_ head as the block to simulate transactions in for estimateGas the `BLOCKHASH` function can't return the hash for the current chain head. So to estimateGas successfully we have to wait until the chain head advances past the L1 head we're specifying in the call.  The dispute game doesn't have that check so doesn't need the wait.